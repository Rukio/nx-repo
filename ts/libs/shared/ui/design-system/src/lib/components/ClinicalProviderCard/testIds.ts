const CLINICAL_PROVIDER_CARD_TEST_IDS = {
  // test id strings will be prefixed by the testIdPrefix prop (ex. xyz-clinical-provider-card)
  CLINICAL_PROVIDER_CARD: 'clinical-provider-card',
  PROVIDER_NAME: 'provider-name',
  PROVIDER_ADDRESS_LINE_1: 'provider-address-line1',
  PROVIDER_CITY_STATE: 'provider-city-state',
  PROVIDER_PHONE_CONTAINER: 'provider-phone-container',
  PROVIDER_PHONE: 'provider-phone',
  PROVIDER_FAX_CONTAINER: 'provider-fax-container',
  PROVIDER_FAX: 'provider-fax',
  PROVIDER_SELECT_BUTTON: 'provider-select-button',
};

export default CLINICAL_PROVIDER_CARD_TEST_IDS;
