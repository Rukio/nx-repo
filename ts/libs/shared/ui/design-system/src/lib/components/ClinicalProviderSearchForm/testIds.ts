const CLINICAL_PROVIDER_SEARCH_FORM_TEST_IDS = {
  FORM_SELECT_BUTTON: 'form-select-button', // with form name suffix, ex. form-select-button-practice
  FIRST_NAME_INPUT: 'first-name-input',
  LAST_NAME_INPUT: 'last-name-input',
  PRACTICE_NAME_INPUT: 'practice-name-input',
  PHARMACY_NAME_INPUT: 'pharmacy-name-input',
  RANGE_IN_MILES_INPUT: 'range-in-miles-input',
  DISTANCE_VALUE: 'distance-value', // with distance as suffix, ex. distance-value-25
  ZIP_CODE_INPUT: 'zip-code-input',
  PHONE_INPUT: 'phone-input',
  INPUT_ERROR: 'error', // with field test id as prefix, ex. last-name-input-error
  SEARCH_BUTTON: 'search-button',
};

export default CLINICAL_PROVIDER_SEARCH_FORM_TEST_IDS;
