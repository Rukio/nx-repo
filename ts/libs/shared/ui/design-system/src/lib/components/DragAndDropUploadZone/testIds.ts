export const DRAG_AND_DROP_TEST_IDS = {
  getDragAndDropUploadZone: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone`,
  getDragAndDropUploadZoneText: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone-text`,
  getDragAndDropZoneUploadErrorText: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone-error-text`,
  getDragAndDropZoneUploadInput: (testIdPrefix: string) =>
    `${testIdPrefix}-drag-and-drop-upload-zone-input`,
};
