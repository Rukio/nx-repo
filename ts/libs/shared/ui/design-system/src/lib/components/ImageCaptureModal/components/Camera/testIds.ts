export const CAMERA_TEST_IDS = {
  getCameraTopText: (testIdPrefix: string) => `${testIdPrefix}-camera-top-text`,
  getCameraBox: (testIdPrefix: string) => `${testIdPrefix}-camera-box`,
  getCameraCanvasBox: (testIdPrefix: string) =>
    `${testIdPrefix}-camera-canvas-box`,
  getCameraBottomText: (testIdPrefix: string) =>
    `${testIdPrefix}-camera-bottom-text`,
  getCameraLoadingIndicator: (testIdPrefix: string) =>
    `${testIdPrefix}-camera-loading-indicator`,
};
