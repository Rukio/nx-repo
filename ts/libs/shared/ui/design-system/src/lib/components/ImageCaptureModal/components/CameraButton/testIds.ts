export const CAMERA_BUTTON_TEST_IDS = {
  getCaptureButton: (testIdPrefix: string) =>
    `${testIdPrefix}-photo-modal-capture-photo-button`,
};
