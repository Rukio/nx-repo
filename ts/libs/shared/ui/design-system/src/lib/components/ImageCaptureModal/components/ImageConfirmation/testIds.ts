export const IMAGE_CONFIRMATION_TEST_IDS = {
  getImageConfirmationTopText: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-top-text`,
  getImageConfirmationImage: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-image`,
  getImageConfirmationImagePlaceholder: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-image-placeholder`,
  getImageConfirmationRetakeButton: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-retake-button`,
  getImageConfirmationButton: (testIdPrefix: string) =>
    `${testIdPrefix}-image-confirmation-button`,
};
