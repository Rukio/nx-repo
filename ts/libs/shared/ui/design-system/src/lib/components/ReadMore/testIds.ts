export const READ_MORE_TEST_IDS = {
  BUTTON: 'read-more-button',
};
