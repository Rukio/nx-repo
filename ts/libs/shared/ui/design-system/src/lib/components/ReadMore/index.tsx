import { FC, ReactNode, useState } from 'react';
import { READ_MORE_TEST_IDS } from './testIds';
import { Box, Link } from '../..';

export type ReadMoreProps = {
  children?: string;
  expandLabel?: ReactNode | string;
  collapseLabel?: ReactNode | string;
  maxTextLength?: number;
};

const ReadMore: FC<ReadMoreProps> = ({
  children = '',
  expandLabel = 'read more',
  collapseLabel = 'read less',
  maxTextLength = 250,
}) => {
  const [showSlicedText, setShowSlicedText] = useState(true);

  const toggleShowSlicedText = () => {
    setShowSlicedText((prev) => !prev);
  };

  const showReadMoreButton = children.length >= maxTextLength;

  const text =
    showSlicedText && showReadMoreButton
      ? `${children?.slice(0, maxTextLength)}...`
      : `${children}`;

  const buttonLabel = showSlicedText ? expandLabel : collapseLabel;

  return (
    <>
      {text}
      {showReadMoreButton && (
        <Box ml={0.5} component="span">
          <Link
            onClick={toggleShowSlicedText}
            component="button"
            data-testid={READ_MORE_TEST_IDS.BUTTON}
            variant="body2"
          >
            {buttonLabel}
          </Link>
        </Box>
      )}
    </>
  );
};

export default ReadMore;
