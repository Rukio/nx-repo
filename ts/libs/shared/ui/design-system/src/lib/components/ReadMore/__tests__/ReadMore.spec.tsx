import ReadMore from '..';
import { render, screen, userEvent, waitFor } from '../../../../test-utils';
import { Basic } from '../__storybook__/ReadMore.stories';
import { READ_MORE_TEST_IDS } from '../testIds';

const containerDataTestId = 'container-test-id';

const mockText =
  'Dolore in irure eu eiusmod adipisicing id laboris culpa aliqua';

const setup = ({
  maxTextLength = 30,
  text = mockText,
}: { maxTextLength?: number; text?: string } = {}) => {
  return render(
    <div data-testid={containerDataTestId}>
      <ReadMore maxTextLength={maxTextLength}>{text}</ReadMore>
    </div>
  );
};

describe('ReadMore', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('snapshots', () => {
    it('should render basic snapshot correctly', () => {
      const { asFragment } = render(<Basic />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('should render text with read more button if text length is more than max', async () => {
    setup();

    const readMoreButton = screen.getByTestId(READ_MORE_TEST_IDS.BUTTON);
    expect(readMoreButton).toBeVisible();
    expect(readMoreButton).toBeEnabled();
  });

  it('should render text without read more button if text length is less than max', async () => {
    setup({ maxTextLength: 100 });

    const readMoreButton = screen.queryByTestId(READ_MORE_TEST_IDS.BUTTON);
    expect(readMoreButton).not.toBeInTheDocument();
  });

  it('should render empty text without read more button if no children was passed', async () => {
    setup({ text: '' });

    const container = await screen.findByTestId(containerDataTestId);
    expect(container).toHaveTextContent('');

    const readMoreButton = screen.queryByTestId(READ_MORE_TEST_IDS.BUTTON);
    expect(readMoreButton).not.toBeInTheDocument();
  });

  it('should expand and collapse text on button click', async () => {
    const slicedText = mockText.slice(0, 30);

    setup();

    const container = await screen.findByTestId(containerDataTestId);
    expect(container).toHaveTextContent(slicedText);

    const readMoreButton = screen.getByTestId(READ_MORE_TEST_IDS.BUTTON);
    expect(readMoreButton).toBeVisible();
    expect(readMoreButton).toBeEnabled();

    await userEvent.click(readMoreButton);

    await waitFor(() => {
      expect(container).toHaveTextContent(mockText);
    });

    await userEvent.click(readMoreButton);

    await waitFor(() => {
      expect(container).toHaveTextContent(slicedText);
    });
  });
});
