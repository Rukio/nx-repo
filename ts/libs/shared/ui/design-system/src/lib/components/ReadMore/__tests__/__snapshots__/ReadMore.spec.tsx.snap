// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMore snapshots should render basic snapshot correctly 1`] = `
<DocumentFragment>
  Qui amet aliqua aliquip qui incididunt dolor. Incididunt amet ea id cillum consequat voluptate ea fugiat aute eiusmod occaecat ut. Sit elit duis irure pariatur minim dolor est voluptate do et tempor dolore voluptate. Est labore elit est voluptate est...
  .emotion-0 {
  margin-left: 4px;
}

.emotion-1 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  color: #0074B8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(0, 116, 184, 0.4);
  position: relative;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.emotion-1:hover {
  text-decoration-color: inherit;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-focusVisible {
  outline: auto;
}

<span
    class="MuiBox-root emotion-0"
  >
    <button
      class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways MuiLink-button emotion-1"
      data-testid="read-more-button"
    >
      read more
    </button>
  </span>
</DocumentFragment>
`;
